File: string/1374_generate_a_string_with_characters_that_have_odd_counts/1374_generate_a_string_with_characters_that_have_odd_counts.py

class Solution(object):
    def generateTheString(self, n):
        if n % 2 == 0 :
            return 'a' * (n-1) + 'b'
        else :
            return 'a' * n 


n = 4 
res = Solution().generateTheString(n)
print(res)

File: string/1374_generate_a_string_with_characters_that_have_odd_counts/1374_generate_a_string_with_characters_that_have_odd_counts.go
package main

import (
	"fmt"
	"strings"
)

func generateTheString(n int) string {
	var rst string
	for i := 0; n != 0 && i < 26; i++ {
		if n%2 == 1 {
			rst += strings.Repeat(string('a'+i), n)
			n = 0
		} else {
			rst += strings.Repeat(string('a'+i), n-1)
			n = 1
		}
	}
	return rst
}

func main() {
	n := 4
	res := generateTheString(n)
	fmt.Println(res)
}
